Be able to get jobs using cibyl from elasticsearch

adrianfusco opened the pull request:
- Modify source registry to add the correct elasticsearch source for queries.
- Add get_jobs method. Now it returns every job it matchs in a 'AttributeDictValue'
- TODO: `get_builds` and `get_tests` methods. Tests.

bregman-arie commented on cibyl/sources/elasticsearch/api.py:
```diff
@@ -81,9 +81,52 @@ def __query_get_hits(self: object, query: dict, index: str = '') -> list:
+            query_body = QueryTemplate('job_name', job_name).get
+
+            builds = self.__query_get_hits(
+                index='jenkins_builds',
```
can this be index=self.index? so users can modify it from configuration

adrianfusco replied:
It depends how we handle it. 

The thing is we’re using in one method one index (`get_jobs` with Jenkins) and in the other method another different index (`get_builds` with jenkins_builds).

bregman-arie replied:
Yeah so probably no. But I wish we didn't have to include indexes names hardcoded in the code

adrianfusco replied:
Maybe we can pass in the configuration file `jobs_index` and `builds_index` and we’ll be able to receive it as you commented. 

I’ll also think another options.

bregman-arie replied:
maybe. but will we need also "tests_index"?
why do we actually have two indexes for builds and jobs? can't we just use one?

adrianfusco replied:
Jobs and builds are in different indexes. The relationship with them is the job name. 
From one side I can get all the jobs from one index. For being able to get the builds I search them by the job name. That's why I need to specify the index for every search.

bregman-arie replied:
Looking at "builds_index" I see for every document the fields: "job_name", "build_result", ... why can't use only "builds_index"?

bregman-arie replied:
k, we can do index optimization later on. Let's merge it when CI passes

adrianfusco replied:
Ok, I'll finish it.

adrianfusco commented:
- Added elasticsearch source in source_factory
- Added multiple query templates:
   * Now we can get one job, multiple jobs, all jobs

Still WIP.

adrianfusco commented:
- Added method `get_last_build`.

adrianfusco commented:
You can review it. There’s another task to add the tests.

adrianfusco commented:
Added:

Tests for ElasticSearchOSP class. Methods: get_builds, get_jobs
Tests for QueryTemplate class.

bregman-arie commented on cibyl/sources/elasticsearch/api.py:
```diff
@@ -15,58 +15,59 @@
 
 
-class ElasticSearchOSP:
+class ElasticSearchOSP(Source):  # pylint: disable=too-few-public-methods
```
Jorge has set "min-public-methods=1" in .pylintrc so you probably don't need to disable it

adrianfusco replied:
You're right. Thx.

bregman-arie reviewed (approved):
Amazing job on this one :)
